### PR TITLE
Initial commit of the ibm-fhir-server helm chart

### PR DESCRIPTION
**Pull request by lmsurpre**
Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

**Review by tbieste — APPROVED** [review]
LGTM

File: ibm-fhir-server/README.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+This helm chart will help you install the IBM FHIR Server in a Kubernetes environment and uses
+ConfigMaps and Secrets to support the wide range of configuration options available for the server.
+
+## Sample usage
+```
+helm upgrade --install ibm-fhir-server . --values values.yaml --set 'ingress.hostname=cluster1-blue-250babbbe4c3000e15508cd07c1d282b-0000.us-east.containers.appdomain.cloud' --set 'ingress.tls[0].secretName=cluster1-blue-250babbbe4c3000e15508cd07c1d282b-0000'
+```
+
+## Prerequisites
+To install the IBM FHIR Server via helm, you must first have a database.
```
Comment (ibmatc) [inline ibm-fhir-server/README.md:16]: Could the default be to use Derby, thus not "requiring" a database to pre-exist?

File: ibm-fhir-server/templates/config-internal.yaml [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+            "enabled": true,
+            "authFilter": {
+              "enabled": false,
+              "authorizedClientCertClientCN": "",
+              "authorizedClientCertIssuerOU": ""
+            }
+          },
+          "oauth": {
+            "enabled": false,
+            "regUrl": "https://<host>:9443/oauth2/endpoint/oauth2-provider/registration",
```
Comment (ibmatc) [inline ibm-fhir-server/templates/config-internal.yaml:84]: looks like this needs to be helm'ized still..?

File: ibm-fhir-server/templates/config.yaml [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+          },
+          "certificates": {
+            "enabled": true,
+            "authFilter": {
+              "enabled": false
+            }
+          },
+          "oauth": {
+            "enabled": false,
+            "regUrl": "https://<host>:9443/oauth2/endpoint/oauth2-provider/registration",
```
Comment (ibmatc) [inline ibm-fhir-server/templates/config.yaml:86]: Again - looks like <host> needs to be helm'ized

**Review by ibmatc — COMMENTED** [review]
LGTM - I might have more comments after I get a chance to use it, but nothing strikes me as wrong/missing for a 0.0.1 version.

File: ibm-fhir-server/templates/config-internal.yaml (hunk 2 above)
Comment (lmsurpre) [reply ibm-fhir-server/templates/config-internal.yaml:84]: yes, we should do this

File: ibm-fhir-server/templates/config.yaml (hunk 3 above)
Comment (lmsurpre) [reply ibm-fhir-server/templates/config.yaml:86]: yes, we should do this

File: ibm-fhir-server/README.md (hunk 1 above)
Comment (lmsurpre) [reply ibm-fhir-server/README.md:16]: the default for the fhir server image is derby, but i was thinking we'd want to encourage users to use a real database.  probably needs further discussion.

**Pull request merged**